server/src/main/java/org/opensearch/index/store/remote/utils/cache/LRUCache.java

```diff
@@ -137,9 +136,7 @@ public V get(K key) {
                 return null;
             }
             // hit
-            if (node.evictable()) {
-                lru.moveToBack(node);
-            }
+            incRef(key);
```

Comment:
I might be missing something here but if we `incRef` on every get and put, wouldn't it mess with the logic of making the node evictable - since every clone operation originally increased the ref?

Reply:
The logic is now like this:

- get from cache (ref count = 1)
- create clone (ref count = 2)
- dec ref (ref count = 1)
- return clone
...
- clone is closed (ref count = 0)


Previously it was:

- get from cache (ref count = 0)
- __!!!!!!!!!!!!!!!!! potential race with eviction !!!!!!!!!!!!!!__
- create clone (ref count = 1)
- return clone
...
- clone is closed (ref count = 0)

Reply:
Ahh. I misread the function name, I thought `put` also did an `incRef` but it was `computeIfPresent`. This makes sense.

Reply:
Sorry. I reread. GH diff is hard. :( 
What about the `put`s - I think there is also `incRef` in there